envoy-control-core/src/main/resources/lua/ingress_client_name_header.lua

```diff
@@ -0,0 +1,25 @@
+function envoy_on_request(handle)
+    local streamInfo = handle:streamInfo()
+    local trusted_header = handle:metadata():get("trusted_client_identity_header") or ""
+    if trusted_header == "" then
+        return
+    end
+
+    if handle:headers():get(trusted_header) ~= nil then
+        handle:headers():remove(trusted_header)
+    end
+
+    if handle:connection():ssl() and streamInfo:downstreamSslConnection() then
```

Comment:
Why are we checking `handle:connection():ssl()` if we're not accessing it?

Reply:
`handle:connection():ssl()` returns nil if this is not a TLS connection. So it's quick way skip the script. Probably checking 
`streamInfo:downstreamSslConnection()` only would be fine.
But in docs there is no mention what is returned by `streamInfo:downstreamSslConnection()` in such case. And `handle:connection():ssl()` is recommended to check if the connection is TLS one.

https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter#ssl
https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter#downstreamsslconnection